File: LabActivitiy/labactivity7.py
# Part A (2 points) - basic Python sorting
def sortIt(listArr):
  listArr.sort()

# empty_list=[ ]
# sortIt(empty_list)      # emptyList will still have the value [ ]
# print(empty_list)

# test_list=[5, -10, 2, -2, 0]
# sortIt(test_list)      # testList will have the value [ -10, -2, 0, 2, 5 ]
# print(test_list)

# Part C (5 points) - Python key selector & custom sorting functions

def shooting_pct(kv):
    attempts = kv.get("shot attempts")
    shots = kv.get("shots made")
    val = shots/attempts
    return val

# shooting_pct({"shot attempts" : 10, "shots made": 5, "minutes": 42}) # would evaluate to 0.5
# shooting_pct({"shot attempts" : 1, "shots made": 1, "minutes": 2}) # would evaluate to 1.0

def shooting_sort(lstArr):
    lstArr.sort(key=shooting_pct)

# nba = [ {"shot attempts" : 184, "shots made": 130, "minutes": 681},
#         {"shot attempts" : 635, "shots made": 271, "minutes": 1059} ]

# shooting_sort(nba) # the order of nba's entries will now be reversed.
# print(nba)

File: LabActivitiy/labactivity7.js
// Part B (2 points) - basic JavaScript sorting

function sortIt(list){
    list.sort(compareFn);
  }
  
  function compareFn(val1, val2){
    if (val1 < val2){
        return -1;
    } else if (val1 > val2) {
        return 1;
    } else{
        return 0;
    }
  }
  
//   let testList=[5, 10, 2, 1, 2, 0];
//   sortIt(testList);
//   console.log(testList);

function rossTrophy(obj1, obj2){
  // Assign "points", "goals", and "games"
  
  if (obj1.points != obj2.points){
    return -1;
  } else if (obj1.points == obj2.points && obj1.goals != obj2.goals){
    return 1;
  } else if (obj1.points == obj2.points && obj1.goals == obj2.goals){
    return 0;
  }
}

let mcdavid = {"points" : 64, "goals": 22, "games": 43};
let draisaitl = {"points" : 64, "goals": 23, "games": 43};
let madeup = {"points" : 24, "goals": 24, "games": 45};

// console.log(rossTrophy(mcdavid, mcdavid)); // would evaluate to 0
// console.log(rossTrophy(draisaitl, mcdavid)); // would evaluate to any Number greater than 0
// console.log(rossTrophy(madeup, mcdavid)); // would evaluate to any Number less than 0

function rossSort(list){
  list.sort(rossTrophy);
}